### PR TITLE
docs: Improve documentation on surface classes

### PR DESCRIPTION
**Pull request by paulgessinger**
In particular `LineSurface` after #2287.

### CODE REVIEW

File: Core/include/Acts/Surfaces/LineSurface.hpp [hunk 1]
```diff
@@ -161,32 +161,26 @@ class LineSurface : public Surface {
...
 
-  /// Specified for LineSurface: global to local method without dynamic
-  /// memory allocation
+  /// Specified for `LineSurface`: global to local method without dynamic
+  /// memory allocation.
   ///
-  /// This method is the true global->local transformation.<br>
-  /// makes use of globalToLocal and indicates the sign of the Acts::eBoundLoc0
-  /// by the given momentum direction
+  /// This method is the true global -> local transformation and  makes use of
```
Comment (andiwand) [inline Core/include/Acts/Surfaces/LineSurface.hpp:167]:
```suggestion
  /// This method is the true global -> local transformation and makes use of
```

File: Core/include/Acts/Surfaces/DiscSurface.hpp [hunk 2]
```diff
@@ -256,7 +256,7 @@ class DiscSurface : public Surface {
   /// @param bcheck The boundary check prescription
   /// @param tolerance the tolerance used for the intersection
   ///
-  ///  <b>mathematical motivation:</b>
+  /// Mathematical motivation:</b>
```
Comment (andiwand) [inline Core/include/Acts/Surfaces/DiscSurface.hpp:259]: opening tag missing now?

```suggestion
  /// <b>Mathematical motivation:</b>
```

File: docs/core/geometry.md [hunk 3]
```diff
@@ -7,187 +7,211 @@ of higher complexity. This design has been chosen as the surface objects can be
...
+members: globalToLocal,localToGlobal,intersect,normal
+---
+:::
+
+### Line surface
+
+{class}`Acts::LineSurface` is a special kind of surface that depends on a reference
+direction, typically the unit momentum direction of a particle. A point in
+space is considered *on surface* if and only if it coincides with the point of
+closest approach between the direction vector $\vec d$ and the line direction
```
Comment (felix-russo) [inline docs/core/geometry.md:132]: I don't understand what the direction vector is, could you please explain?

File: docs/core/geometry.md [hunk 4]
```diff
@@ -7,187 +7,211 @@ of higher complexity. This design has been chosen as the surface objects can be
...
+
 ## Layer classes
 
-The `Layer` class is an extension of the `Surface` class that allows the
+
+The {class}`Acts::Layer` class is an extension of the {class}`Acts::Surface` class that allows the
 definition of sub surfaces (sensitive surfaces for modules, or extra material
 surfaces).
 
 The Layer can simply correspond to a 'virtual' surface in the detector
```
Comment (felix-russo) [inline docs/core/geometry.md:167]:
```suggestion
The layer can simply correspond to a 'virtual' surface in the detector
```

File: docs/core/geometry.md [hunk 5]
```diff
@@ -7,187 +7,211 @@ of higher complexity. This design has been chosen as the surface objects can be
...
 description or represent a more complex object that may contain:
 
-* a representing surface, which is accessible via a `representingSurface()`
-* method an array of contained surfaces, accessible via `surfaceArray()` method
+* a representing surface, which is accessible via a {func}`Acts::Layer::surfaceRepresentation`
+* method an array of contained surfaces, accessible via {func}`Acts::Layer::surfaceArray` method
```
Comment (felix-russo) [inline docs/core/geometry.md:171]: I don't understand this bullet point (there is some grammatical problem I think)

File: docs/core/geometry.md [hunk 6]
```diff
@@ -7,187 +7,211 @@ of higher complexity. This design has been chosen as the surface objects can be
...
 
-The following illustration shows an x-y view of a cylinder layer with planar
+The following illustration shows an $xy$ view of a cylinder layer with planar
 detection modules:
 
 ![CylinderLayer](/figures/geometry/CylinderLayer.png)
 
 Modules can be sorted onto layer using all supported binning methods described
-through the `SurfaceArray` class, the binning can be adjusted to fit as good as
+through the {class}`Acts::SurfaceArray` class, the binning can be adjusted to fit as good as
```
Comment (felix-russo) [inline docs/core/geometry.md:181]:
```suggestion
through the {class}`Acts::SurfaceArray` class. The binning can be adjusted to fit as well as
```

File: docs/core/geometry.md [hunk 7]
```diff
@@ -7,187 +7,211 @@ of higher complexity. This design has been chosen as the surface objects can be
...
 possible.
 
 ![DiscLayerEB](/figures/geometry/DiscLayerEB.png)
 
 The un-occupied space in a volume which contains a layer array is filled with
```
Comment (felix-russo) [inline docs/core/geometry.md:186]:
```suggestion
The unoccupied space in a volume that contains a layer array is filled with
```

File: docs/core/geometry.md [hunk 8]
```diff
@@ -7,187 +7,211 @@ of higher complexity. This design has been chosen as the surface objects can be
...
-The `Volume` class is a container of `BoundarySurface` objects, where each
-`BoundarySurface` is an extension of the `Surface` class with additional
-information about the attached Volumes. The normal vector of the surface
-defines an *inside* (opposite w.r.t. the normal vector) and an *outside* (along
-w.r.t. the normal vector) direction. Either a single volume or an array of
-volumes can be attached to a volume.
+The {class}`Acts::Volume` class is a container of
+{type}`Acts::BoundarySurface` objects, where each
+{type}`Acts::BoundarySurface` is an extension of the {class}`Acts::Surface`
+class with additional information about the attached Volumes. The normal vector
```
Comment (felix-russo) [inline docs/core/geometry.md:201]:
```suggestion
class with additional information about the attached volumes. The normal vector
```

File: docs/core/geometry.md [hunk 9]
```diff
@@ -205,35 +229,40 @@ The basic information for any material is:
...
-This information is confined together in the `Material` class.
+This information is confined together in the {class}`Acts::Material` class.
 
 ```{note}
-In track reconstruction, only an effective material description is needed, i.e. non-physical values in regards of the atomic number, the elementary charge or even the density are allowed, as long as the effective relative radiation length and `A/Z x rho` ratio can be retrieved.
-This enables compatictification of the material description, as the element composition record does not have to be kept.
+In track reconstruction, only an effective material description is needed, i.e.
+non-physical values in regards of the atomic number, the elementary charge or
+even the density are allowed, as long as the effective relative radiation
+length and $A/Z \times \rho$ ratio can be retrieved.  This enables compatictification
```
Comment (felix-russo) [inline docs/core/geometry.md:238]:
```suggestion
length and $A/Z \times \rho$ ratio can be retrieved.  This enables the compactification
```

File: docs/core/geometry.md [hunk 10]
```diff
@@ -205,35 +229,40 @@ The basic information for any material is:
...
+of the material description, as the element composition record does not have to
+be kept.
 ```
 
 Surface based material extends this material information by representative
```
Comment (felix-russo) [inline docs/core/geometry.md:243]:
```suggestion
Surface based material extends this material information by a representative
```

File: docs/core/geometry.md [hunk 11]
```diff
@@ -205,35 +229,40 @@ The basic information for any material is:
...
-thickness, the corresponding object is called `MaterialSlab`. The
+thickness, the corresponding object is called {class}`Acts::MaterialSlab`. The
```
Comment (felix-russo) [inline docs/core/geometry.md:244]:
```suggestion
thickness; the corresponding object is called {class}`Acts::MaterialSlab`. The
```

File: Core/include/Acts/Surfaces/LineSurface.hpp [hunk 12]
```diff
@@ -161,32 +161,26 @@ class LineSurface : public Surface {
...
+  /// The measurement plane is determined by the two orthogonal vectors @f$
+  /// \vec{\texttt{measY}} = \vec{\texttt{Acts::eBoundLoc1}} @f$ and @f$
+  /// \vec{\texttt{measX}} = \vec{\texttt{measY}} \times
+  /// \frac{\vec{p}}{|\vec{p}|} @f$.
+  ///
+  /// The sign of the radius (@f$ d_{0} @f$ ) is then defined by the projection
+  /// of @f$ \vec{d} @f$ onto @f$ \vec{measX} @f$:<br> @f$ sign = -sign(\vec{d}
+  /// \cdot \vec{measX}) @f$
   ///
   /// @image html figures/SignOfDriftCircleD0.gif
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply Core/include/Acts/Surfaces/LineSurface.hpp:185]: 🤔

File: docs/core/geometry.md [hunk 13]
```diff
@@ -7,187 +7,211 @@ of higher complexity. This design has been chosen as the surface objects can be
...
 ## Surface classes
 
-The `Surface` class builds the core class of all geometry objects and can be
-used natively with the propagation and extrapolation modules. The common
-`Surface` virtual base defines the public interface of all surfaces. The
-different concrete `Surface` classes are defined by their respective native
-local coordinate system, while different shapes on surfaces are defined by
-`SurfaceBounds` classes which every surface must provide. In case of boundless
-surfaces, a special `InfiniteBounds` class is available.
+All classes which represent thin surface in ACTS inherit from
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply docs/core/geometry.md:57]: I guess this wording refers to other geometry objects like volumes.

File: docs/core/geometry.md (hunk 3 above)
Comment (paulgessinger) [reply docs/core/geometry.md:132]: It's just the momentum direction vector that is mentioned earlier in the paragraph.

File: docs/core/geometry.md (hunk 5 above)
Comment (paulgessinger) [reply docs/core/geometry.md:171]:
```suggestion
* an array of contained surfaces, accessible via {func}`Acts::Layer::surfaceArray` method
```

File: Core/include/Acts/Surfaces/LineSurface.hpp [hunk 14]
```diff
@@ -161,32 +161,26 @@ class LineSurface : public Surface {
...
-  /// \f$.<br>
-  ///
-  /// The sign of the radius (\f$ d_{0} \f$ ) is then defined by the projection
-  /// of
-  /// \f$ \vec{d} \f$
-  /// onto \f$ \vec{measX} \f$:<br>
-  /// \f$ sign = -sign(\vec{d} \cdot \vec{measX}) \f$
+  /// state, then @f$ \vec d @f$ lies within the so called measurement plane.
+  /// The measurement plane is determined by the two orthogonal vectors @f$
+  /// \vec{\texttt{measY}} = \vec{\texttt{Acts::eBoundLoc1}} @f$ and @f$
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply Core/include/Acts/Surfaces/LineSurface.hpp:177]: I think this is supposed to refer to the local coordinate axis vector.

[the remark this answers is not in the document]
Comment (felix-russo) [reply Core/include/Acts/Surfaces/LineSurface.hpp:177]: I would write 
```suggestion
  /// \vec{\texttt{measY}} = \vec{e}_z @f$ and @f$
```
then

File: docs/core/geometry.md (hunk 3 above)
Comment (felix-russo) [reply docs/core/geometry.md:128]: I would then write 
`... typically the unit momentum direction $\vec d$ of a particle.`
to make things even clearer

File: Core/include/Acts/Surfaces/LineSurface.hpp (hunk 12 above)
[the remark this answers is not in the document]
Comment (paulgessinger) [reply Core/include/Acts/Surfaces/LineSurface.hpp:185]: Strange, seems to be there when I build it locally...

File: Core/include/Acts/Surfaces/CylinderSurface.hpp [hunk 15]
```diff
@@ -38,7 +38,7 @@ class DetectorElementBase;
 /// since it builds the surfaces of all TrackingVolumes at container level
 /// for a cylindrical tracking geometry.
 ///
-/// @image html figures/CylinderSurface.png
+/// @image html /figures/CylinderSurface.png
```
Comment (felix-russo) [inline Core/include/Acts/Surfaces/CylinderSurface.hpp:41]: I still don't see the image unfortunately... 
image

**Review by felix-russo — COMMENTED** [review]
Otherwise good to go

File: Core/include/Acts/Surfaces/CylinderSurface.hpp (hunk 15 above)
Comment (andiwand) [reply Core/include/Acts/Surfaces/CylinderSurface.hpp:41]: @paulgessinger do you know what the problem could be? I would have thought that doxygen would crash if it does not find an image link

Comment (paulgessinger) [reply Core/include/Acts/Surfaces/CylinderSurface.hpp:41]: It's rewriting them a bunch of times when copying them around, so not sure.

Comment (andiwand) [reply Core/include/Acts/Surfaces/CylinderSurface.hpp]: looks like this was already broken https://acts.readthedocs.io/en/latest/api/class/class_acts_1_1_cylinder_surface.html?highlight=cylindersurface

**Review by andiwand — DISMISSED** [review]

**Comment by andiwand** [comment]
@paulgessinger if you like we can merge this already and fix the images afterwards since it seems to be an unrelated problem

**Comment by paulgessinger** [comment]
I'm just doing some more CI debugging

**Comment by paulgessinger** [comment]
Ok, dead end for right now. Let's merge.

**Review by felix-russo — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6051939)!

|status|job|
|------|---|
|:red_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753531) |
|:red_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753530) |
|:red_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753529) |
|:red_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753528) |
|:red_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753527) |
|:red_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753526) |
|:red_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753525) |
|:red_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753524) |
|:red_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753523) |
|:red_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753522) |
|:red_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753521) |
|:red_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753520) |
|:red_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753519) |
|:red_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753518) |
|:red_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753517) |
|:red_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753516) |
|:red_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31753515) |

**Comment by paulgessinger** [comment]
Failure with a missing symbol. This seems weird, I'm investigating.

**Comment by paulgessinger** [comment]
Indeed technical error.

**Pull request merged**